mlir/lib/Conversion/AIRToAIEPass.cpp

```diff
@@ -3425,13 +3424,34 @@ class AIRLinalgOpToLibraryCallRewrite
     if (fnName.empty())
       return failure();
 
+    // Function to get operands of the library call that will
+    // replace the given linalg op.
+    auto getLibFnOperands = [](linalg::LinalgOp op) {
+      SmallVector<Value> operands;
+      for (auto operand : op->getOperands()) {
+        if (auto operation = operand.getDefiningOp()) {
+          if (dyn_cast<memref::ReshapeOp>(operation) ||
+              dyn_cast<memref::ExpandShapeOp>(operation) ||
+              dyn_cast<memref::CollapseShapeOp>(operation)) {
```

Comment:
Could use `isa<X...>(operation)>` here or `isa_and_present` to combine with line 3432:
```C++
Operation *operation = operand.getDefiningOp();
if (!isa_and_present<memref::ReshapeOp, memref::ExpandShapeOp, memref::CollapseShapeOp>(operation)>)
  continue;
operands.push_back(operation->getOperand(0));
```